File: 20. Valid Parentheses/20. Valid Parentheses.go
func isValid(s string) bool {
    stack := make([]rune, 0)
    for _, c := range s {
        if c == '(' || c == '{' || c == '[' {
            stack = append(stack, c)
        } else {
            if len(stack) == 0 {
                return false
            }
            if (c == ')' && stack[len(stack)-1] != '(') ||
               (c == '}' && stack[len(stack)-1] != '{') ||
               (c == ']' && stack[len(stack)-1] != '[') {
                return false
            }
            stack = stack[:len(stack)-1]
        }
    }
    return len(stack) == 0
}

File: 20. Valid Parentheses/20. Valid Parentheses.dart
class Solution {
  bool isValid(String s) {
    List<String> stack = [];
    for (int i = 0; i < s.length; i++) {
      if (s[i] == '(' || s[i] == '{' || s[i] == '[') {
        stack.add(s[i]);
      } else {
        if (stack.isEmpty) return false;
        if ((s[i] == ')' && stack.last != '(') ||
            (s[i] == '}' && stack.last != '{') ||
            (s[i] == ']' && stack.last != '[')) {
          return false;
        }
        stack.removeLast();
      }
    }
    return stack.isEmpty;
  }
}